include/net/buf.h

```diff
@@ -818,7 +818,7 @@ struct net_buf_pool {
 
 #if defined(CONFIG_NET_BUF_POOL_USAGE)
 	/** Amount of available buffers in the pool. */
-	int16_t avail_count;
+	atomic_t avail_count;
```

Comment:
Doesn't it make sense to move the `avail_count` before `uninit_count` or after `pool_size` so that you don't increase the struct size due to padding?

Reply:
I can change it after pool_size, but not sure it will help, because it also depend what the size of "struct k_lifo free" which is the first variable on this struct.

Reply:
No, you're right - I should have looked at the entire struct layout. No need to change anything here.